sdk/src/main/java/com/bugsnag/android/SessionTracker.java

```diff
@@ -0,0 +1,233 @@
+package com.bugsnag.android;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Pair;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+class SessionTracker implements Application.ActivityLifecycleCallbacks {
```

Comment:
Looks like this also handles Lifecycle Breadcrumbs as well as session tracking? This could be made more obvious with comments / rename of class